src/utils.ts

```diff
@@ -102,14 +102,14 @@ export namespace Utils {
    * @param {string} challengeTx SEP0010 challenge transaction in base64.
    * @param {string} serverAccountID The server's stellar account (public key).
    * @param {string} networkPassphrase The network passphrase, e.g.: 'Test SDF Network ; September 2015'.
-   * @param {string} [homeDomain=undefined] The field is reserved for future use and not used.
-   * @returns {Transaction|string} The actual submited transaction and the stellar public key (master key) used to sign the Manage Data operation.
+   * @param {string} [homeDomain=undefined] The home domain that should be included in the Manage Data operation's string key.
+   * @returns {Transaction|string} The actual submitted transaction and the stellar public key (master key) used to sign the Manage Data operation.
    */
   export function readChallengeTx(
     challengeTx: string,
     serverAccountID: string,
     networkPassphrase: string,
-    _homeDomain?: string,
+    homeDomain?: string,
```

Comment:
❗ I think we could make this field non-optional. It's not optional in some of our other SDKs, and we want client to use it. What do you think @quietbits @JakeUrban?

Reply:
If `homeDomain` is required, then we should make it required here. It wouldn't break anything, I assume?

Reply:
It should not be optional, SEP-10 3.0 mandates that the Manage Data key is validated so SDK users shouldn't be allowed to skip it in this SDK version.